mlfinlab/structural_breaks/cusum.py

```diff
@@ -0,0 +1,68 @@
+"""
+Implementation of Chu-Stinchcombe-White test
+"""
+
+import pandas as pd
+import numpy as np
+from mlfinlab.util import mp_pandas_obj
+
+
+def _get_s_n_for_t(series: pd.Series, test_type: str, molecule: list) -> pd.Series:
+    """
+    Get maximum S_n_t value for each value from molecule for Chu-Stinchcombe-White test
+    :param series: (pd.Series) to get statistics for
+    :param test_type: (str): two-sided or one-sided test
+    :param molecule: (list) of indices to get test statistics for
+    :return: (pd.Series) of statistics
+    """
+
+    s_n_t_series = pd.DataFrame(index=molecule, columns=['stat', 'critical_value'])
+    for index in molecule:
+
+        series_t = series.loc[:index]
+        squared_diff = series_t.diff().dropna() ** 2
+        integer_index = series_t.index.get_loc(index)
+        sigma_sq_t = 1 / (integer_index - 1) * sum(squared_diff)
+
+        max_s_n_value = -np.inf
+        max_s_n_critical_value = None  # Corresponds to c_aplha[n,t]
+
+        # Indices difference for the last index would yield 0 -> division by zero warning,
+        # no need to index last value iteration
+        for ind in series_t.index[:-1]:
+            if test_type == 'one_sided':
+                values_diff = series.loc[index] - series.loc[ind]
+            elif test_type == 'two_sided':
+                values_diff = abs(series.loc[index] - series.loc[ind])
+            else:
+                raise ValueError('Test type is unknown: can be either one_sided or two_sided')
+
+            temp_integer_index = series_t.index.get_loc(ind)
+            s_n_t = 1 / (sigma_sq_t * np.sqrt(integer_index - temp_integer_index)) * values_diff
+            if s_n_t > max_s_n_value:
+                max_s_n_value = s_n_t
+                max_s_n_critical_value = np.sqrt(4.6 + np.log(integer_index - temp_integer_index))
```

Comment:
Please, document what 4.6 is and it would be perfect to define it as CONST.

Reply:
What is 4.6?

Reply:
This is the empirical constant from the book. Let me comment on that